cloudbuild.yaml

```diff
@@ -0,0 +1,32 @@
+# See https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md for more details on image pushing process
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 1200s
+# this prevents errors if you don't use both _GIT_TAG and _PULL_BASE_REF,
+# or any new substitutions added in the future.
+options:
+  substitution_option: ALLOW_LOOSE
+steps:
+  # The image must contain bash and curl. Ideally it should also contain
+  # the desired version of Go (currently defined in release-tools/travis.yml),
+  # but that just speeds up the build and is not required.
+  - name: 'gcr.io/k8s-testimages/gcb-docker-gcloud:v20200421-a2bf5f8'
+    entrypoint: bash
+    env:
+    - GIT_TAG=${_GIT_TAG}
+    - PULL_BASE_REF=${_PULL_BASE_REF}
+    - HOME=/root
+    args:
+    - .cloudbuild.sh
+substitutions:
+  # _GIT_TAG will be filled with a git-based tag for the image, of the form vYYYYMMDD-hash, and
+  # can be used as a substitution
+  _GIT_TAG: '12345'
+  # _PULL_BASE_REF will contain the ref that was pushed to trigger this build -
+  # a branch like 'master' or 'release-0.2', or a tag like 'v0.2'.
+  _PULL_BASE_REF: 'master'
+artifacts:
+  objects:
+    location: 'gs://k8s-artifacts-csi/dev'
+    paths: 'bin/csi-proxy-${_PULL_BASE_REF}.exe'
```

Comment:
Ah sorry one more thought came to me. Do we want to have a "csi-proxy/bin/" directory in case in the future we have more than one binary we want to put here?

Reply:
you mean the gs bucket location as gs://k8s-artifacts-csi/csi-proxy/bin? Or put it in paths: csi-proxy/bin/csi-proxy-${_PULL_BASE_REF}.exe? 

For the bucket, if there are other binaries, they will have different names? For the paths here, I am not sure adding csi-proxy is useful?

Reply:
either way. I think it's better organized when each binary is in separate directories, especially when we start to get many versions stored.

Reply:
got it. But I thought paths is the local path where the file is stored locally, and location is the remote store where the file will be uploaded. So  we should add it to location?
How about this gs://k8s-artifacts-csi/csi-proxy/dev?